### PR TITLE
Rohan i18nify studio client application

### PR DESCRIPTION
**Pull request by rohank07**
### Intent

> Describe briefly what problem this pull request resolves, or what new feature it introduces. Include screenshots of any new or altered UI. Link to any Github issues that are related. 

This PR tackles hard-coded strings to i18n-ify in i18n-ify org.rstudio.studio.client & .application (excluding false-positives).
Issue: https://github.com/StatCan/daaas/issues/715

### Approach

> Describe the approach taken and the tradeoffs involved if non-obvious; add an overview of the solution if it's complicated.

This pull-request and future ones will tackle i18nify classes at package-level (e.g. org.rstudio.core) instead of the per-component approach (like we did for [all-options-menu](https://github.com/rstudio/rstudio/pull/9599))

Implementation follows the same pattern from #9485 as described in gwt/tools/i18n-helpers/README.md. Running the create_dev_locale.sh will create all the new "dev" locale from any new "en" properties files for easier visual confirmation. Refer to README for more information.
### Automated Tests

> Indicate whether this change includes unit tests or integration tests, or link a work item or pull request to add those tests to another repo. If the change cannot or will not be covered by a test, indicate why.

### QA Notes

> Add additional information for QA on how to validate the change, paying special attention to the level of risk, adjacent areas that could be affected by the change, and any important contextual information not present in the linked issues. 

Modify an entry in one of the new _en.properties files, and refresh RStudio with ant in devmode.
Run create_dev_locale.sh to clone all _en properties files, which will add a prefix of "@" to each constant and message. Then open RStudio using http://localhost:8787/?locale=dev and browse to the Options->* menus. You can also view in desktop mode by swapping the locale element value attribute from value="en" to value="dev" in RStudio.gwt.xml.

### Checklist

- [ ] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [ ] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [ ] This PR passes all local unit tests

cc @gtritchie

### CODE REVIEW

**Comment by rohank07** [comment]
Hello @gtritchie, experiencing some build issues with this PR. Encountering issues with classes and subclasses that exist in `/client/application/ui`. I might be missing the source folders in `RStudio.gwt.xml`.  I would appreciate it if you can give me some additional insight. 

image

File: src/gwt/src/org/rstudio/studio/client/RStudioGinModule.java [hunk 1]
```diff
@@ -23,18 +24,14 @@
 import org.rstudio.core.client.VirtualConsole;
 import org.rstudio.core.client.VirtualConsole.PreferencesImpl;
 import org.rstudio.core.client.VirtualConsoleFactory;
+import org.rstudio.studio.client.application.*;
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/RStudioGinModule.java:27]: We generally don't use wildcard imports, please revert back to individual ones.

See: "Avoid using wildcard (org.path.to.package.*) imports" on this page: https://github.com/rstudio/rstudio/wiki/Coding-Standards-(java,-cpp)

File: src/gwt/src/org/rstudio/studio/client/RStudioGinModule.java [hunk 2]
```diff
@@ -528,4 +525,5 @@ protected void configure()
    {
       bind(Widget.class).annotatedWith(Names.named(name)).to(clazz);
    }
+   //private final StudioClientApplicationConstants constants_ = GWT.create(StudioClientApplicationConstants.class);
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/RStudioGinModule.java:528]: Not used, remove.

File: src/gwt/src/org/rstudio/studio/client/RStudioGinModule.java [hunk 3]
```diff
@@ -14,6 +14,7 @@
  */
 package org.rstudio.studio.client;
 
+import com.google.gwt.core.client.GWT;
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/RStudioGinModule.java:17]: Not used, remove

File: src/gwt/src/org/rstudio/studio/client/application/Application.java [hunk 4]
```diff
@@ -249,14 +249,14 @@ else if (error.getCode() == ServerError.LICENSE_USAGE_LIMIT)
                // we display a special dialog for license limit issues
                // to allow the user to attempt to re-launch the session
                globalDisplay_.showGenericDialog(GlobalDisplay.MSG_ERROR,
-                                              "Licensing Limit Reached",
+                                              constants_.licensingLimitCaption(),
                                               error.getUserMessage() + "\n\n" +
-                                                 "Please quit any unused running sessions and try again, " +
-                                                 "or contact your administrator to update your license.",
+                                                 constants_.quitRunningSessionsMessage() +
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/application/Application.java:254]: Use a single string, not concatenation.

File: src/gwt/src/org/rstudio/studio/client/application/Application.java [hunk 5]
```diff
@@ -249,14 +249,14 @@ else if (error.getCode() == ServerError.LICENSE_USAGE_LIMIT)
...
+                                                 constants_.contactAdminMessage(),
                                               buttonLabels, elementIds, buttonOperations, 0);
             }
-            else if (StringUtil.equals(error.getUserMessage(), "Unable to connect to service") ||
-                     StringUtil.equals(error.getUserMessage(), "Error occurred during transmission"))
+            else if (StringUtil.equals(error.getUserMessage(), constants_.unableToConnectMessage()) ||
+                     StringUtil.equals(error.getUserMessage(), constants_.errorTransmissionMessage()))
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/application/Application.java:259]: This code suggests logic that depends on exact string matches. Would need to do deeper review to see if this is a concern here. Leave this comment here so I remember to take another look.

File: src/gwt/src/org/rstudio/studio/client/application/Application.java [hunk 6]
```diff
@@ -268,29 +268,29 @@ else if (StringUtil.equals(error.getUserMessage(), "Unable to connect to service
...
+               buttonLabels.add(constants_.retryButtonLabel());
                elementIds.add(ElementIds.DIALOG_RETRY_BUTTON);
                buttonOperations.add(() ->
                {
                   Window.Location.reload();
                });
                globalDisplay_.showGenericDialog(GlobalDisplay.MSG_ERROR,
-                                              "Cannot Connect to R Session",
-                                              "Could not connect to the R session on RStudio Server.\n\n" +
+                                              constants_.cannotConnectRCaption(),
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/application/Application.java:286]: Single string.

File: src/gwt/src/org/rstudio/studio/client/application/ApplicationInterrupt.java [hunk 7]
```diff
@@ -219,20 +220,20 @@ public void onConsoleBusy(ConsoleBusyEvent event)
    private void showInterruptUnresponsiveDialog()
    {
       showTerminationDialog(
-         "R is not responding to your request to interrupt processing so " +
-         "to stop the current operation you may need to terminate R entirely." +
+         constants_.terminationDialog() +
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/application/ApplicationInterrupt.java:223]: Make this a single string.

File: src/gwt/src/org/rstudio/studio/client/application/ApplicationInterrupt.java [hunk 8]
```diff
@@ -304,9 +305,9 @@ private void finishInterrupt(InterruptHandler handler)
...
    private final ApplicationServerOperations server_;
    private final ErrorManager errorManager_;
-   
-   private final static String TERMINATION_CONSEQUENCE_MSG = 
-      "Terminating R will cause your R session to immediately abort. " +
-      "Active computations will be interrupted and unsaved source file " +
-      "changes and workspace objects will be discarded.";
+   private static final StudioClientApplicationConstants constants_ = GWT.create(StudioClientApplicationConstants.class);
+   private final static String TERMINATION_CONSEQUENCE_MSG =
+      constants_.terminationConsequenceMessage() +
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/application/ApplicationInterrupt.java:310]: Should be a single string.

File: src/gwt/src/org/rstudio/studio/client/application/ApplicationQuit.java [hunk 9]
```diff
@@ -146,7 +146,7 @@ else if (workbenchContext_.isServerBusy() && terminalHelper_.warnBeforeClosing(b
             globalDisplay_.showYesNoMessage(
                   MessageDialog.QUESTION,
                   caption, 
-                  msg + " Are you sure you want to quit?",
+                  msg + " " + constants_.applicationQuitMessage(),
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/application/ApplicationQuit.java:149]: For flexibility in localization, would run this through a placeholder string, e.g.:

{0} Are you sure you want to quit?

File: src/gwt/src/org/rstudio/studio/client/application/StudioClientApplicationConstants.java [hunk 10]
```diff
@@ -0,0 +1,1823 @@
+package org.rstudio.studio.client.application;
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/application/StudioClientApplicationConstants.java:1]: Requires our standard file header comment block.

File: src/gwt/src/org/rstudio/studio/client/application/model/ProductEditionInfo.java [hunk 11]
```diff
@@ -15,13 +15,15 @@
...
+import com.google.gwt.core.client.GWT;
+import org.rstudio.studio.client.application.StudioClientApplicationConstants;
 import org.rstudio.studio.client.application.Desktop;
 
 public class ProductEditionInfo
 {
    public String editionName()
    {
-      return "RStudio" + (Desktop.isDesktop() ? "" : " Server");
+      return "RStudio" + (Desktop.isDesktop() ? "" : " " + constants_.rStudioEditionName());
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/application/model/ProductEditionInfo.java:26]: Use string substitution, e.g. "RStudio{0}"

File: src/gwt/src/org/rstudio/studio/client/application/ui/ApplicationWindow.java [hunk 12]
```diff
@@ -157,10 +159,10 @@ public void showApplicationUpdateRequired()
    {
       globalDisplay_.showMessage(
             GlobalDisplay.MSG_INFO,
-            "Application Updated",
-            "An updated version of RStudio is available. Your browser will " +
-            "now be refreshed with the new version. All current work and data " +
-            "will be preserved during the update.",
+              constants_.applicationUpdatedCaption(),
+            constants_.applicationUpdatedMessage() +
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/application/ui/ApplicationWindow.java:163]: Use single string, not three

File: src/gwt/src/org/rstudio/studio/client/application/ui/ApplicationWindow.java [hunk 13]
```diff
@@ -272,10 +274,9 @@ public void hideWarning()
    public void showSessionAbendWarning()
    {
       globalDisplay_.showErrorMessage(
-            "R Session Error",
-            "The previous R session was abnormally terminated due to " +
-            "an unexpected crash.\n\n" +
-            "You may have lost workspace data as a result of this crash.");
+            constants_.rSessionErrorCaption(), constants_.previousRSessionsMessage() +
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/application/ui/ApplicationWindow.java:277]: Use single string, not concatenation.

File: src/gwt/src/org/rstudio/studio/client/application/ui/RequestLogDetail.java [hunk 14]
```diff
@@ -43,11 +45,11 @@ public RequestLogDetail(RequestLogEntry entry)
       panel.getElement().getStyle().setOverflow(Overflow.AUTO);
 
       HTML html = new HTML();
-      html.setText("Request ID: " + entry.getRequestId() + "\n\n"
-                   + "== REQUEST ======\n"
+      html.setText(constants_.requestIdText() + entry.getRequestId() + "\n\n"
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/application/ui/RequestLogDetail.java:48]: This creates a string used for debug diagnostics, don't localize any of this.

File: src/gwt/src/org/rstudio/studio/client/application/ui/RequestLogVisualization.java [hunk 15]
```diff
@@ -107,15 +109,15 @@ public void onClick(ClickEvent event)
...
-      instructions_.setHTML("<p>Click on a request to see details. Click on the " +
-                            "background to show these instructions again.</p>" +
-                            "<h4>Available commands:</h4>" +
-                            "<ul>" +
-                            "<li>Esc: Close</li>" +
-                            "<li>P: Play/pause</li>" +
-                            "<li>E: Export</li>" +
-                            "<li>I: Import</li>" +
-                            "<li>+/-: Zoom in/out</li>" +
+      instructions_.setHTML("<p>"+ constants_.clickRequestText() +
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/application/ui/RequestLogVisualization.java:112]: Debug diagnostics feature, prefer we don't localize this.

File: src/gwt/src/org/rstudio/studio/client/application/ui/appended/ApplicationEndedPopupPanel.java [hunk 16]
```diff
@@ -58,28 +59,28 @@ public static void showMultiSessionQuit()
...
       asyncShow(SUICIDE, description, null);
    }
 
    public static void showDisconnected()
    {
       String description =
-         "This browser was disconnected from the R session because another " +
-         "browser connected (only one browser at a time may be connected " +
-         "to an RStudio session). You may reconnect using the button below.";
+         constants_.browserDisconnectedMessage() +
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/application/ui/appended/ApplicationEndedPopupPanel.java:72]: Make a single string

File: src/gwt/src/org/rstudio/studio/client/application/ui/appended/ApplicationEndedPopupPanel.java [hunk 17]
```diff
@@ -58,28 +59,28 @@ public static void showMultiSessionQuit()
...
 
       asyncShow(DISCONNECTED, description, null);
    }
 
    public static void showOffline()
    {
       String description =
-         "RStudio is temporarily offline due to system maintenance. We " +
-         "apologize for the inconvenience, please try again in a few minutes.";
+         constants_.rStudioOfflineMessage() +
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/application/ui/appended/ApplicationEndedPopupPanel.java:82]: Make a single string

**Review by gtritchie — CHANGES_REQUESTED** [review]

**Comment by gtritchie** [comment]
Awesome, I will re-review this and the other PRs tomorrow.

**Review by gtritchie — APPROVED** [review]
Looks good, thanks!

**Pull request merged**